src/test/java/teammates/test/pageobjects/StudentProfilePage.java

```diff
@@ -158,6 +159,10 @@ private void ensureGenderIsSelectedAs(String gender) {
     public void uploadPicture() {
         uploadPictureSubmit.click();
         browser.selenium.waitForPageToLoad("10000");
+        
+        // Intended compounded explicit wait due to .click()'s inconsistency in triggering selenium's
+        // waitForPageToLoad. Do not remove, can cause a higher rate of random failure in this test.
+        ThreadHelper.waitFor(1000);
```

Comment:
I remember now, I removed it and did a undo because I wanted to look at the comment and forgot to remove it at the end when I commited it, no wonder the test felt like it was going 5 seconds slower than usual!